src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -525,5 +610,10 @@ public static byte[] EncodeLiteralHeaderFieldWithoutIndexingToAllocatedArray(int
                 span = new byte[span.Length * 2];
             }
         }
+
+        public static byte ToLowerAscii(char c)
+        {
+            return (byte)((uint)(c - 'A') <= ('Z' - 'A') ? c | 0x20 : c);
```

Comment:
```suggestion
            return (byte)(c | 0x20);
```
No need for the comparison.